fix: reduce external library logging noise

diegomrsantos opened the pull request:
## Issue Addressed

Excessive logging noise from external libraries libp2p_gossipsub and discv5 affecting log readability and storage.

## Proposed Changes

- Set external library log levels to WARN in libp2p_discv5_layer filter
- Changed from DEBUG to `EnvFilter::new("libp2p_gossipsub=warn,discv5=warn")`
- Added documentation explaining the noise reduction strategy

## Additional Info

Preserves Anchor's debug logging while filtering external library noise. Targeted change affecting only the specialized logging layer.

diegomrsantos committed:
fix: reduce external library logging noise by 95%

- Change libp2p_gossipsub and discv5 log levels from DEBUG to WARN
- Preserves Anchor's console (INFO) and file (DEBUG) logging architecture
- Targets identified high-frequency noise: 62K+ logs/hour to ~3K logs/hour
- Maintains environment variable override capability via from_env_lossy()

Impact: Reduces log storage from projected 474GB/year to ~25GB/year

diegomrsantos committed:
perf: use string literal instead of format! for static filter

Addresses clippy warning about useless format! usage

dknopik commented:
A lot of the "[Penalty]" logmessages in libp2p are on `debug` level. IMO these are quite useful for debugging in scenarios where Anchor unexpectedly blocks a lot of peers. wdyt?

diegomrsantos commented:
> A lot of the "[Penalty]" logmessages in libp2p are on `debug` level. IMO these are quite useful for debugging in scenarios where Anchor unexpectedly blocks a lot of peers. wdyt?

They can be quite noisy, can't they? I'm actually thinking those should be trace and we should make a PR. Wdyt?

dknopik commented:
Sure, they can be noisy, but also very useful for troubleshooting, especially as they represent an abnormal state. I think they are fine.

diegomrsantos commented:
> Sure, they can be noisy, but also very useful for troubleshooting, especially as they represent an abnormal state. I think they are fine.

I think you're right. So maybe the current content of libp2p should be actually `trace`:
  - "Message already received, ignoring" - routine deduplication
  - "Sending message to peer" - low-level message routing
  - "Completed forwarding message" - internal state transitions
  - "Handling message from peer" - message processing steps
  
Wdyt?

dknopik commented:
While I am inclined to agree, I am not sure what the `rust-libp2p` guidelines WRT logging are. So I guess João might help here.

diegomrsantos commented:
> While I am inclined to agree, I am not sure what the `rust-libp2p` guidelines WRT logging are. So I guess João might help here.

@jxs what are your thoughts?

jxs commented:
Hi folks, thanks for pinging and sorry for the delay.
I am not sure if I understand the question, but, there aren't repo wide logging guidelines in libp2p that I am aware of, each Behaviour has its own. And I think putting discovery and gossipsub to `warn` might suppress useful information

diegomrsantos committed:
fix: enhance libp2p/discv5 logging with target info and refined filtering

- Add target field to log format for better correlation with filters
- Replace simple external library filter with granular approach
- Set default to WARN level for external libraries to reduce noise
- Preserve DEBUG logging for critical gossipsub peer_score and gossip_promises modules
- Add fallback to debug level if filter parsing fails

diegomrsantos commented on anchor/src/main.rs:
```diff
@@ -205,12 +205,14 @@ pub fn enable_logging(
-                        EnvFilter::builder()
-                            .with_default_directive(Level::DEBUG.into())
-                            .from_env_lossy(),
+                        EnvFilter::try_new("warn,libp2p_gossipsub::peer_score=debug,libp2p_gossipsub::gossip_promises=debug")
```
@dknopik let me know what you think

diegomrsantos replied:
@dknopik I tried to test it with the impostor mode. Didn't see any logs on discv5 and libp2p as the node doesn't seem to find any peers.

diegomrsantos committed:
Merge branch 'release-v0.3.0' into fix/reduce-external-library-logging-noise

diegomrsantos commented:
> Hi folks, thanks for pinging and sorry for the delay. I am not sure if I understand the question, but, there aren't repo wide logging guidelines in libp2p that I am aware of, each Behaviour has its own. And I think putting discovery and gossipsub to `warn` might suppress useful information

Hi @jxs, sorry for the lack of context. We were talking about the content of this file: [libp2p.txt](https://github.com/user-attachments/files/22238585/libp2p.txt) in this message https://github.com/sigp/anchor/pull/583#issuecomment-3258364766

dknopik reviewed (approved):
LGTM
